tests/phpunit/tests/blocks/renderCommentTemplate.php

```diff
@@ -512,4 +512,97 @@ public function test_rendering_comment_template_unmoderated_preview() {
 			'Should not include any unapproved comments after removing filter'
 		);
 	}
+
+	public function test_rendering_comment_template_sets_comment_id_context() {
+		$parsed_comment_author_name_block = parse_blocks( '<!-- wp:comment-author-name /-->' )[0];
+		$comment_author_name_block        = new WP_Block(
+			$parsed_comment_author_name_block,
+			array(
+				'commentId' => self::$comment_ids[0],
+			)
+		);
+		$comment_author_name_block_markup = $comment_author_name_block->render();
+
+		$render_block_callback = static function( $block_content, $block ) use ( $parsed_comment_author_name_block ) {
+			// Insert a Comment Author Name block (which requires `commentId`
+			// block context to work) after the Comment Content block.
+			if ( 'core/comment-content' !== $block['blockName'] ) {
+				return $block_content;
+			}
+
+			$inserted_content = render_block( $parsed_comment_author_name_block );
+			return $inserted_content . $block_content;
+		};
+
+		add_filter( 'render_block', $render_block_callback, 10, 3 );
+		$parsed_blocks = parse_blocks(
+			'<!-- wp:comment-template --><!-- wp:comment-content /--><!-- /wp:comment-template -->'
+		);
+		$block         = new WP_Block(
+			$parsed_blocks[0],
+			array(
+				'postId' => self::$custom_post->ID,
+			)
+		);
+		$markup        = $block->render();
+		remove_filter( 'render_block', $render_block_callback );
+
+		$this->assertStringContainsString( $comment_author_name_block_markup, $markup );
+	}
+
+	public function test_inner_block_inserted_by_render_block_data_is_retained() {
+		$render_block_callback = new MockAction();
+		add_filter( 'render_block', array( $render_block_callback, 'filter' ), 10, 3 );
+
+		$render_block_data_callback = static function( $parsed_block ) {
+			// Add a Social Links block to a Comment Template block's inner blocks.
+			if ( 'core/comment-template' === $parsed_block['blockName'] ) {
+				$inserted_block_markup = <<<END
+<!-- wp:social-links -->
+<ul class="wp-block-social-links"><!-- wp:social-link {"url":"https://wordpress.org","service":"wordpress"} /--></ul>
+<!-- /wp:social-links -->'
+END;
+
+				$inserted_blocks = parse_blocks( $inserted_block_markup );
+
+				$parsed_block['innerBlocks'][] = $inserted_blocks[0];
+			}
+			return $parsed_block;
+		};
+
+		// Remove auto-insertion filter so it won't collide.
+		remove_filter( 'render_block_data', 'gutenberg_auto_insert_child_block' );
```

Comment:
This refers to a `gutenberg` filter it looks like, should this be removed?

Reply:
Ah, looks like these might be leftovers from an experiment over in https://github.com/WordPress/gutenberg/pull/51294 ? The tests pass for me fine locally with these lines removed for this PR.

Reply:
https://github.com/WordPress/wordpress-develop/pull/4507/commits/c6257da39005ea91363287b4507669b7432c6688